daemon/response.go

```diff
@@ -79,7 +79,7 @@ type respJSON struct {
 	Type       ResponseType `json:"type"`
 	Status     int          `json:"status-code"`
 	StatusText string       `json:"status"`
-	Result     interface{}  `json:"result"`
+	Result     interface{}  `json:"result,omitempty"`
```

Comment:
```
<zyga-solus> pedronis: can you please look at http://paste.ubuntu.com/26029227
<zyga-solus> pedronis: is that a sane assertion (syntax wise)
<zyga-solus> pedronis: this is what we fail on that EOF bug
<pedronis> zyga-solus: it looks like one, I don't the problem is assertion
<pedronis> zyga-solus: my guess would be the change in response,  but I'm not sure why it would trigger problems
<pedronis> zyga-solus: easiest would be to get the branch, try one of the failing tests, try to undo that change, see if it changes something, then figure out what's happening
<zyga-solus> pedronis: right, I'm looking at the diff again but nothing stands out
<zyga-solus> pedronis: the only new thing is Tracks 
<zyga-solus> pedronis: it didn't have any annotation before
<pedronis> but that struct is not involved with ack at all
<pedronis> as I said afaict the only logical candidate
<zyga-solus> hmm
<pedronis> is the change in response.go
<pedronis> it would be good to verify that theory
<zyga-solus> pedronis: sure, that's easy
<zyga-solus> thank you
<zyga-solus> pedronis: so, it's not that 
<zyga-solus> pedronis: rebuilt snap, snapd, reinstalled, restarted snapd
```

So unless I missed something, it is not this line.

Reply:
Heh, so obviously re-exec bit me.

```
<pedronis> zyga-solus: actually I'm quite sure it is
<zyga-solus> pedronis: ok, let me re-check 
<pedronis> I mean I�see the now the code
<pedronis> why that would matter
<pedronis> we do things like this in client:   err := jsonutil.DecodeWithNumber(bytes.NewReader(rsp.Result), v)
<pedronis> json.Unmarshal(rsp.Result, &resultErr)
<zyga-solus> ok
<zyga-solus> I definitely removed that change
<pedronis> and this is the field in the client:          Result     json.RawMessage `json:"result"`
<zyga-solus> and rebuilt
* zyga-solus wonders
<pedronis> so if the field is not there (instead of null), I suspect those to break with EOF
<zyga-solus> and restarted
<zyga-solus> Chipaca: woot, thank you!
<zyga-solus> pedronis: could it be a combination, I'm sure I removed this aspect of the patch
<zyga-solus> pedronis: and new snapd is runing
<zyga-solus> *running
<zyga-solus> ah
<zyga-solus> sorry, obviously 
<zyga-solus> reexec
<pedronis> heh
<zyga-solus> thank you
<zyga-solus> :)
* pstolowski rozłączył/a się (Quit: Leaving)
<pedronis> now we need to decide whether we want to fix client or we consider this an incompatible change
<pedronis> maybe Chipaca or niemeyer have opinions on that
<zyga-solus> yes, confirmed
<zyga-solus> funny outcome 
<pedronis> with how client is written we cannot omit result
```